pages/blog/posts/understanding-lexical-dynamic-scopes.md

```diff
@@ -0,0 +1,244 @@
+subschemas are the same. 
+
+In the following example, the lexical and dynamic scopes for the subschema
+highlighted in red align:
+
+![A simple example where lexical and dynamic scopes align](/img/posts/2024/understanding-lexical-dynamic-scopes/aligned.webp)
+
+### References and the Evaluation Path
+
+References are the only exception to top down evaluation. By definition,
+references (both static and dynamic) make the evaluation process break from
+this mode by <i>jumping</i> to another location (as we saw in the evaluation path
+figure at the beginning of the Dynamic Scope section). As a consequence, when
+the evaluation process encounters a reference, lexical scopes and dynamic
+scopes will typically diverge.
+
+To recap, for the lexical scope, following a reference involves abandoning the
+lexical scope of the reference schema and entering the lexical scope of the
+destination schema. In comparison, for the dynamic scope, <i>following a
+reference involves retaining the current dynamic scope, and adding the
+destination schema to it</i>.
+
+Consider the following example of a schema that follows a reference to an
+anchor. In the left, the lexical scope of the destination schema (highlighted
+in red), consists of it parent schema (highlighted in green), and its
+grandparent schema (highlighted in blue). However, in the right, the dynamic
+scope of the destination schema (highlighted in red) consists of the schema
+that referenced the anchor (highlighted in green) and the schema that led to
+the evaluation of such reference (highlighted in blue):
+
+![Comparison of lexical scope and dynamic scope with a reference](/img/posts/2024/understanding-lexical-dynamic-scopes/references-dynamic.webp)
+
+### Beyond Schema Resource Boundaries
+
+In comparison to the lexical scope, the dynamic scope of a schema is not
+limited by its schema resource. Moreover, the dynamic scope is a singleton
+entity that remains for the entirety of the evaluation process. In other words,
+the dynamic scope <i>is</i> the complete evaluation path.
+
+Consider a variation of the previous example, where this time the anchor is
+within a subschema of a different schema resource. The lexical scope of the
+destination schema drastically changes as it hits its schema resource barrier,
+where as the dynamic scope of the destination schema remains the same:
+
+![Comparison of lexical scope and dynamic scope through schema resources](/img/posts/2024/understanding-lexical-dynamic-scopes/references-dynamic-resource.webp)
+
+Finally, note that the dynamic scope not only persists through local references
+to other schema resources, but across remote references as well.
+
+Summary
+-------
+
+Understanding how static and dynamic scopes work is essential for gaining a
+deeper understanding of JSON Schema. The most important points to remember are
+summarized in the following table:
+
+| Comparison Point                     | Lexical Scope | Dynamic Scope |
```

Comment:
It'll be interesting to see this table rendered on the web.  It's pretty wide.

Reply:
This is how it looks like. I think its OK, so I'll resolve this thread, but let me know if not!

<img width="1021" alt="Screenshot 2024-01-22 at 11 22 37" src="https://github.com/json-schema-org/website/assets/2192773/c9836314-4745-4d60-b998-89f7a96df4e8">